src/schemas/migration.ts

```diff
@@ -13,8 +14,50 @@ export const executeSchemaMigration = async ({
   schemaName: string
   migrations: Migrations
 }): Promise<any> => {
+  const migrationSource = buildMigrationSource({ schemaName, migrations })
+
+  return knex.migrate
+    .latest({
+      schemaName,
+      migrationSource,
+    })
+    .catch((error) => {
+      return error
+    })
+}
+
+export const executeSchemaMigrationFromDir = async ({
+  knex,
+  schemaName,
+  directory,
+}: {
+  knex: Knex
+  schemaName: string
+  directory: string
+}): Promise<any> => {
+  let fileNames
+  try {
+    fileNames = await promises.readdir(directory)
+  } catch {
+    return `Could not read directory "${directory}"`
+  }
+
+  const mgs: Migrations = {}
+
+  for (const filename of fileNames) {
```

Comment:
You could separate this routine into a new function. It could be something called: `buildMigrationSourceFromFiles`, maybe?